scripts/kernel_bump.sh

```diff
@@ -0,0 +1,225 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright (C) 2024 Olliver Schinagl <oliver@schinagl.nl>
+
+set -eu
+if [ -n "${DEBUG_TRACE_SH:-}" ] && \
+   [ "${DEBUG_TRACE_SH:-}" != "${DEBUG_TRACE_SH#*"$(basename "${0}")"*}" ] || \
+   [ "${DEBUG_TRACE_SH:-}" = 'all' ]; then
+	set -x
+fi
+
+REQUIRED_COMMANDS='
+	[
+	basename
+	command
+	echo
+	exit
+	git
+	printf
+	set
+	shift
+'
+
+_msg()
+{
+	_level="${1:?Missing argument to function}"
+	shift
+
+	if [ "${#}" -le 0 ]; then
+		echo "${_level}: No content for this message ..."
+		return
+	fi
+
+	echo "${_level}: ${*}"
+}
+
+e_err()
+{
+	_msg 'err' "${*}" >&2
+}
+
+e_warn()
+{
+	_msg 'warning' "${*}"
+}
+
+e_notice()
+{
+	_msg 'notice' "${*}"
+}
+
+usage()
+{
+	echo "Usage: ${0}"
+	echo 'Helper script to bump the target kernel version, whilst keeping history.'
+	echo "    -p  Optional Platform name (e.g. 'ath79' [PLATFORM_NAME]"
+	echo "    -s  Source version of kernel (e.g. 'v6.1' [SOURCE_VERSION])"
+	echo "    -t  Target version of kernel (e.g. 'v6.6' [TARGET_VERSION]')"
+	echo
+	echo 'All options can also be passed in environment variables (listed between [BRACKETS]).'
+	echo 'Example: scripts/kernel_bump.sh -p realtek -s v6.1 -t v6.6'
+}
+
+cleanup()
+{
+	trap - EXIT HUP INT QUIT ABRT ALRM TERM
+
+	if [ -n "${initial_branch:-}" ] && \
+	   [ "$(git rev-parse --abbrev-ref HEAD)" != "${initial_branch:-}" ]; then
+		git switch "${initial_branch}"
+	fi
+}
+
+init()
+{
+	initial_branch="$(git rev-parse --abbrev-ref HEAD)"
+	initial_commitish="$(git rev-parse HEAD)"
+
+	trap cleanup EXIT HUP INT QUIT ABRT ALRM TERM
+}
+
+do_source_target()
+{
+	_target_dir="${1:?Missing argument to function}"
+	_op="${2:?Missing argument to function}"
+
+}
+
+bump_kernel()
+{
+	platform_name="${platform_name##*'/'}"
+
+	if [ -z "${platform_name:-}" ]; then
+		platform_name="${PWD##*'/'}"
+	fi
+
+	if [ "${PWD##*'/'}" = "${platform_name}" ]; then
+		_target_dir='./'
+	else
+		_target_dir="target/linux/${platform_name}"
+	fi
+
+	if [ ! -d "${_target_dir}/image" ]; then
+		e_err 'Cannot find target linux directory.'
+		exit 1
+	fi
+
+	git switch --force-create '__openwrt_kernel_files_mover'
+
+	for _path in "${_target_dir}/"*; do
+		if [ ! -s "${_path}" ] || \
+		   [ "${_path}" = "${_path%%"-${source_version}"}" ]; then
+			continue
+		fi
+
+		_target_path="${_path%%"-${source_version}"}-${target_version}"
+		if [ -s "${_target_path}" ]; then
+			e_err "Target '${_target_path}' already exists!"
+			exit 1
+		fi
+
+		git mv \
+			"${_path}" \
+			"${_target_path}"
+	done
+
+	find "${_target_dir}" -iname "config-${source_version}" | while read -r _config; do
+		_path="${_config%%"/config-${source_version}"}"
+		git mv "${_config}" "${_path}/config-${target_version}"
+	done
+
+	git commit \
+		--signoff \
```

Comment:
There is no need for sign-off here.  Since these are generated by a script, rather than typing into a shell this cannot be copyrighted.

Reply:
Why not? You are the author of this change, just because a script did it, you still take ownership of the change.

I think this is a slippery slope, what if AI wrote 5 commits, I would still need to relinquish ownership.
What if I use a script to change all spaces to tabs, and make the script fancy to make the commits also. What if I did the git commit work manually, due to a bug with the script and my target. Should I then omit it?

So maybe there's not a 'hard-need', but it certainly does not hurt.

Reply:
Issue is this **cannot** by copyrighted.  The sole human generated input is the arguments given to the script.  One would be laughed out of Court for trying to assert copyright over running a script with 2 version numbers (which come from elsewhere) and a device name (which comes from the repository).  I'm a bit worried about implicitly claiming copyright, when it cannot be.  I am not a lawyer, but _to me_ this seems worrisome.

Reply:
But what are you copyrighting? The code or the 'move'. (e.g. the action performed by a human using a script.

So if I executes the commands manually that the script would, I also don't have to sign-off the commits I just? I don't see the difference between me running the commands, or me, through a script. I still did the work, I just had a little help :)

I still hold responsibility here in the end.

Reply:
Read about copyright.

The term I've run across is "sweat of the brow".  If you manually type in each command by hand, then yes there could be some degree of copyright claim.  Typing the 3 arguments to this script is simply too little work.